doc/api/networking.rst

```diff
@@ -192,10 +192,10 @@ DHCPv4
 .. doxygengroup:: dhcpv4
    :project: Zephyr
 
-MQTT 3.1.1
-==========
+MQTT 3.1.1 over Sockets
```

Comment:
Do we want to keep referring to this as MQTT over sockets, also when the old implementation is going to be removed? Can we just call this MQTT and drop the socket from the name of the doxygen group and folders?

Reply:
I think we can drop the `over Sockets` part, just added it to be consistent with CoAP. @jukkar what do you think?

Reply:
Yes, we should drop the socket from various names etc. Can I propose that we remove the legacy MQTT first, and then do these renames etc in separate PR?